### PR TITLE
fix: Fix messy code cased by jsonDecode

### PR DESCRIPTION
**Pull request by yang-chao**

### CODE REVIEW

**Comment by yang-chao** [comment]
Http response text messy code like this

File: lib/src/core/networking/client.dart [hunk 1]
```diff
@@ -29,8 +29,9 @@ class OpenAINetworkingClient {
         "request to $from finished with status code ${response.statusCode}");
 
     OpenAILogger.log("starting decoding response body");
+    Utf8Decoder utf8decoder = Utf8Decoder();
```
Comment (anasfik) [inline lib/src/core/networking/client.dart:32]: Great

File: lib/src/core/networking/client.dart [hunk 2]
```diff
@@ -29,8 +29,9 @@ class OpenAINetworkingClient {
...
     final Map<String, dynamic> decodedBody =
-        jsonDecode(response.body) as Map<String, dynamic>;
+        jsonDecode(utf8decoder.convert(response.bodyBytes)) as Map<String, dynamic>;
```
Comment (anasfik) [inline lib/src/core/networking/client.dart:34]: Greay!

**Comment by anasfik** [comment]
Hi [yang-chao](https://github.com/yang-chao), Thank you very much for this contribution and the notice of this issue.

LGTM!

**Pull request merged**